std/array.d

```diff
@@ -380,6 +380,113 @@ if (isNarrowString!String)
     )));
 }
 
+/**
+ * Returns a static array from the stack with length of `s` and initializes
+ * it with copies of the elements of range $(D r). The remainder of the
+ * array is default-initialized.
+ *
+ * Recommended particulary for complex initializations of static arrays
+ * during compilation.
+ *
+ * Params:
+ *      r = The range (or aggregate with $(D opApply) function) whose elements
+ *      are copied into the stack-allocated array. Must be as long or shorter
+ *      than `s`, otherwise causes an array overflow.
+ *      rangeLength = outputs the amount of elements copied. Same as
+ *      $(D r).$(REF walkLength, std,range,primitives).
+ * Returns:
+ *      An initialized static array
+ */
+ForeachType!Range[s] array(size_t s, Range)(Range r, out size_t rangeLength)
+if (isInputRange!Range && !isInfinite!Range)
+{
+    alias E = ForeachType!Range;
+
+    import std.conv : emplaceRef;
+    import std.algorithm.mutation : uninitializedFill;
+
+    rangeLength = 0;
+    if (__ctfe)
+    {
+        // Compile-time version to avoid memcpy calls.
+        Unqual!E[s] result;
+        foreach (e; r)
+        {
+
```

Comment:
Please no spurious empty lines. This particular one separates nothing in particular and adds negative readability value.